core/src/main/scala/com/stripe/dagon/DependantGraph.scala

```diff
@@ -1,57 +0,0 @@
-/*
- Copyright 2013 Twitter, Inc.
- Copyright 2017 Stripe, Inc.
-
- Licensed under the Apache License, Version 2.0 (the "License");
- you may not use this file except in compliance with the License.
- You may obtain a copy of the License at
-
- http://www.apache.org/licenses/LICENSE-2.0
-
- Unless required by applicable law or agreed to in writing, software
- distributed under the License is distributed on an "AS IS" BASIS,
- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- See the License for the specific language governing permissions and
- limitations under the License.
- */
-
-package com.stripe.dagon
-
-import Graphs._
-
-/**
- * Given Dag and a List of immutable nodes, and a function to get
- * dependencies, compute the dependants (reverse the graph)
- */
-abstract class DependantGraph[T] {
-  def nodes: List[T]
-  def dependenciesOf(t: T): Iterable[T]
-
-  lazy val allTails: List[T] = nodes.filter { fanOut(_).get == 0 }
-  private lazy val nodeSet: Set[T] = nodes.toSet
-
-  /**
-   * This is the dependants graph. Each node knows who it depends on
-   * but not who depends on it without doing this computation
-   */
-  private lazy val graph: NeighborFn[T] = reversed(nodes)(dependenciesOf(_))
-
-  private lazy val depths: Map[T, Int] = dagDepth(nodes)(dependenciesOf(_))
-
-  /**
-   * The max of zero and 1 + depth of all parents if the node is the graph
-   */
-  def isNode(p: T): Boolean = nodeSet.contains(p)
-  def depth(p: T): Option[Int] = depths.get(p)
```

Comment:
can we add this to ExpressionDag if we remove this here? We aren't using it, but summingbird and scalding to use this code, so removing it I think is a bummer.